runtime/cudaq/qis/managers/default/DefaultExecutionManager.cpp

```diff
@@ -86,6 +86,25 @@ class DefaultExecutionManager : public cudaq::BasicExecutionManager {
     requestedAllocations.clear();
   }
 
+  void initializeState(const std::vector<cudaq::QuditInfo> &targets,
+                       const cudaq::SimulationState *state) override {
+    // Note: a void* ptr doesn't provide enough info to the simulators, hence
+    // need a dedicated code path.
+    // TODO: simplify/combine the two code paths (raw vector and state).
+    if (!requestedAllocations.empty() &&
+        targets.size() != requestedAllocations.size()) {
+      assert(targets.size() < requestedAllocations.size());
+      const auto numDefaultAllocs =
+          requestedAllocations.size() - targets.size();
+      simulator()->allocateQubits(numDefaultAllocs);
+      // The targets will be allocated in a specific state.
+      simulator()->allocateQubits(targets.size(), state);
```

Comment:
Do these two allocations need to be grouped back together in any way? Does the order of the allocations here matter?

I'm thinking about computing offsets into a "register/array/vector" of qubits.

Reply:
No, currently we don't group them back in any way. These calls will flow to simulator backends as `addQubitsToState` calls to grow the state. 
We preserve the order of `qvector` allocation while supporting batching of default qvector allocations, e.g.

```
qvector q1(2); // <- qubit id 0, 1 in the simulator
qvector q2(2); // <- qubit id 2, 3 in the simulator
qvector q3(state); // the above allocation requests for a total of 4 qubits in |0000> state were batched -> need to flush before adding some qubits in a specific state, e.g., qubits 4, 5 (if the state is a 2-qubit state).
```
`qvector` holds a `qudits` vector to track these indices as well.